ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -651,13 +657,114 @@ def virtual_router_mac_address(self):
 
     @cached_property
     def enable_trunk_groups(self):
-        return get(self._hostvars, "enable_trunk_groups", default=False)
+        if self._any_network_services:
+            return get(self._hostvars, "enable_trunk_groups", default=False)
+        return None
+
+    @cached_property
+    def only_local_vlan_trunk_groups(self):
+        if self.enable_trunk_groups:
+            return get(self._hostvars, "only_local_vlan_trunk_groups", default=False)
+        return None
+
+    @cached_property
+    def _endpoint_vlans_and_trunk_groups(self):
+        '''
+        Return list of vlans and list of trunk groups used by connected_endpoints
+        '''
+        if not (self._any_network_services and self.connected_endpoints):
+            return [], []
+
+        vlans = []
+        trunk_groups = []
+
+        port_profiles = get(self._hostvars, "port_profiles", default=[])
+        # Support legacy data model by converting nested dict to list of dict
+        port_profiles = self._convert_dicts(port_profiles, "profile")
+
+        connected_endpoints_keys = get(self._hostvars, "connected_endpoints_keys", default=[])
+        # Support legacy data model by converting nested dict to list of dict
+        connected_endpoints_keys = self._convert_dicts(connected_endpoints_keys, "key")
+        for connected_endpoints_key in connected_endpoints_keys:
+            connected_endpoints_key_key = connected_endpoints_key.get("key")
+            if connected_endpoints_key_key is None or get(self._hostvars, connected_endpoints_key_key) is None:
+                # Invalid connected_endpoints_key.key. Skipping.
+                continue
+
+            connected_endpoints = get(self._hostvars, connected_endpoints_key_key)
+            # Support legacy data model by converting nested dict to list of dict
+            connected_endpoints = self._convert_dicts(connected_endpoints, 'name')
+            for connected_endpoint in connected_endpoints:
+                for adapter in connected_endpoint.get("adapters", []):
+                    profile_name = adapter.get("profile")
+                    adapter_profile = get_item(port_profiles, "profile", profile_name, default={})
+                    parent_profile_name = adapter_profile.get("parent_profile")
+                    parent_profile = get_item(port_profiles, "profile", parent_profile_name, default={})
+
+                    adapter_settings = self._combine(parent_profile, adapter_profile, adapter, recursive=True, list_merge='replace')
+
+                    if self.hostname not in adapter_settings.get("switches", []):
+                        # This switch is not connected to this endpoint. Skipping.
+                        continue
+
+                    if "vlans" in adapter_settings and adapter_settings["vlans"] not in ["all", "", None]:
+                        vlans.extend(map(int, self._range_expand(adapter_settings["vlans"])))
```

Comment:
One of my topologies failed here due to `vlans` in my combined adapter settings being an int, not a string or list, which caused range_expand to choke on it.
```
ansible.errors.AnsibleFilterError: value must be of type list or str, got <class 'int'>
```
Easy fix would be:

```suggestion
                        vlans.extend(map(int, self._range_expand(str(adapter_settings["vlans"]))))
```
But I'm not sure if want to support lists of ranges in something as simple as a VLAN range.